metron-platform/metron-common/src/main/java/org/apache/metron/common/cli/ConfigurationManager.java

```diff
@@ -130,29 +201,39 @@ public void pull(CuratorFramework client, String outFileStr, final boolean force
       public void visit(ConfigurationType configurationType, String name, String data) {
         File out = getFile(outputDir, configurationType, name);
         if (!out.exists() || force) {
-          if(!out.exists()) {
+          if (!out.exists()) {
             out.getParentFile().mkdirs();
           }
           try {
             Files.write(data, out, Charset.defaultCharset());
           } catch (IOException e) {
             throw new RuntimeException("Sorry, something bad happened writing the config to " + out.getAbsolutePath() + ": " + e.getMessage(), e);
           }
-        }
-        else if(out.exists() && !force) {
+        } else if (out.exists() && !force) {
           throw new IllegalStateException("Unable to overwrite existing file (" + out.getAbsolutePath() + ") without the force flag (-f or --force) being set.");
         }
       }
     });
   }
 
   public void push(String inputDirStr, CuratorFramework client) throws Exception {
-      final File inputDir = new File(inputDirStr);
+    final File inputDir = new File(inputDirStr);
```

Comment:
The examples in the test plan don't work directly because they NPE here.  I assume it's because inputDirStr is null here and we only check for validity after wrapping it in a `File`.  While you're in here, could you add a check (and meaningful exception) for the case where inputDirStr is null?

Error is:
```
[root@node1 zookeeper]# $METRON_HOME/bin/zk_load_configs.sh -z $ZOOKEEPER -m PUSH -c GLOBAL
Exception in thread "main" java.lang.NullPointerException
	at java.io.File.<init>(File.java:277)
	at org.apache.metron.common.cli.ConfigurationManager.push(ConfigurationManager.java:230)
	at org.apache.metron.common.cli.ConfigurationManager.run(ConfigurationManager.java:256)
	at org.apache.metron.common.cli.ConfigurationManager.run(ConfigurationManager.java:242)
	at org.apache.metron.common.cli.ConfigurationManager.main(ConfigurationManager.java:350)
```

Reply:
That was actually me missing the "-i" parameter in the test script. I've modified the test accordingly. In general we should be better guarding user inputs, e.g. some may have noticed that help never actually worked without throwing an exception. I could add a check there also. I'm inclined to rewrite our configuration classes altogether as I think they can be greatly simplified.

Reply:
Yeah, I think we could definitely handle the scripts better in general.

I'm not opposed to just leaving this as-is, but it seems like a decent / small opportunity to make it more usable. Up to you.

Reply:
Latest commit addresses the input dir null check.